index.bs

```diff
@@ -6003,6 +6007,36 @@ leakage due to such an attack:
         to the preceding authentication step, where it may be easier to solve.
 
 
+## Privacy leak via credential IDs ## {#sctn-credential-id-privacy-leak}
+
+[INFORMATIVE]
+
+This privacy consideration applies to [=[RPS]=] supporting [=single-factor=] authentication with [=non-resident credentials=].
+
+In this case the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument risks leaking [PII],
+if the user can initiate an [=authentication ceremony=] by only providing a username.
+[=Credential IDs=] are designed to not be correlatable between [=[RPS]=],
+but the length of a [=credential ID=] might be a hint as to what type of [=authenticator=] created it.
+It is likely that a user will use the same username and set of [=authenticators=] for several [=[RPS]=],
+so the number of [=credential IDs=] in {{PublicKeyCredentialRequestOptions/allowCredentials}} and their lengths
+might serve as a global correlation handle to de-anonymize the user.
+
+The simplest ways to prevent this information leak
+is to not support [=single-factor=] authentication with [=non-resident credentials=],
+for example by:
```

Comment:
This suggestion -- while simple to describe -- harms interop across the ecosystem. Some authenticators don't work at some sites. That is a poor trade-off for many.

Reply:
This is not about blocking some authenticators, but recommending against some use cases. Non-resident authenticators are perfectly fine to use for second-factor authentication.

Reply:
We should not recommend against this uses case: accepting authenticators with non-resident credentials as 1st factors.
Using a phone or laptop with a builtin platform authenticator, plus having a Security Key for bootstrap & backup, is a great combination that we should encourage.

Reply:
Well, you can't have it both ways. Either you're leaking information via `allowCredentials`, or you don't support exposing `allowCredentials` without some form of preceding authentication step (password or cookie, for example).

Reply:
"single-factor" is wrong. The info leak also occurs for a Security Key that does multi-factor auth with a non-resident key.

Reply:
Hm, you're right. Good catch, thanks!

Reply:
Addressed in bba65aa